script.js

```diff
@@ -0,0 +1,146 @@
+// Possible states of the game
+const GameStates = {
+    INPROGRESS: "In Progress",
+    XVICTORY: "X wins",
+    OVICTORY: "O wins",
+    TIE: "Tie"
+}
+let gameState = GameStates.INPROGRESS
+
+const board = document.querySelectorAll('.grid-square')
+
+// Possible states of a square
+const SquareStates = {
+    X: "X",
+    O: "O",
+    EMPTY: "Empty"
+}
+
+const isTaken = (squareId) => {
+    return isX(squareId) || isO(squareId)
+}
+
+const isX = (squareId) => {
+    return !document.getElementById('ximg' + squareId).classList.contains('hidden')
+}
+
+const isO = (squareId) => {
+    return !document.getElementById('oimg' + squareId).classList.contains('hidden')
+}
+
+const getSquareStates = (squareId) => {
+    if (isX(squareId)){
+        return SquareStates.X
+    }
+    else if (isO(squareId)){
+        return SquareStates.O
+    }
+    else {
+        return SquareStates.EMPTY
+    }
+}
+
+const getTurn = () => {
+    turn = 0
+    for (let i = 0; i < board.length; i++){
+        if (isTaken(i)){
+            turn++
+        }
+    }
+    return turn
+}
+
+const updateTurn = () => {
+    if (gameState !== GameStates.INPROGRESS) {
+        return
+    }
+    let turn = getTurn() % 2 === 0 ? "X" : "O"
+    document.getElementById("turntxt").innerHTML = "It is " + turn + "'s turn."
+}
+
+const checkTie = () => {
+    if (gameState !== GameStates.INPROGRESS) {
+        return
+    }
+    let movesAvailable = false
+    for (let i = 0; i < board.length && !movesAvailable; i++){
+        movesAvailable = !isTaken(i)
+    }
+    if (!movesAvailable) { 
+        console.log('Tie!')
+        gameState = GameStates.TIE
+    }
+}
+
+const checkVictory = (squareA, squareB, squareC) => {
+    if (gameState !== GameStates.INPROGRESS) {
+        return
+    }
+    if (getSquareStates(squareA) !== SquareStates.EMPTY &&
+        getSquareStates(squareA) === getSquareStates(squareB) &&
+        getSquareStates(squareB) === getSquareStates(squareC)) {
+        console.log(getSquareStates(squareA) + ' wins!')
+        gameState = isX(squareA) ? GameStates.XVICTORY : GameStates.OVICTORY
+    }
+}
+
+const evaluateBoard = () => {
+    checkVictory(0, 1, 2) // top row
```

Comment:
Could you put these combos into an array and loop over it, instead of having all the separate function calls?